Engine/Src/Ancona/Framework/Screens/AbstractScreen.cpp

```diff
@@ -23,7 +23,7 @@ AbstractScreen::AbstractScreen(
 
 AbstractScreen::~AbstractScreen()
 {
-    if (_requestList != nullptr)
+    if (_requestList != nullptr && _clearResourcesOnExit)
```

Comment:
This is leaking the resources. If we need to defer cleanup, we should implement a "Vacuum" operation that loops over the resources and deletes unused references. That allows us to differ the operation to after the next request list is loaded in.